.github/workflows/ci.yml

```diff
@@ -89,3 +89,41 @@ jobs:
                   name: Screenshots-${{ matrix.tag }}-${{ matrix.browser }}
                   path: screenshots/
                   if-no-files-found: error
+
+    test-package:
+
+        needs: [pre-commit]
+
+        strategy:
+            matrix:
+                tag: [latest]
+                python-version: ['3.8', '3.9', '3.10']
+            fail-fast: false
+
+        runs-on: ubuntu-latest
+        timeout-minutes: 30
+
+        steps:
+
+            - name: Check out app
+              uses: actions/checkout@v3
+
+            - name: Set up Python
+              uses: actions/setup-python@v4
+              with:
+                  python-version: ${{ matrix.python-version }}
+                  cache: pip
+                  cache-dependency-path: |
+                      **/setup.cfg
+                      **/pyproject.toml
+                      **/requirements*.txt
+
+            - name: Install package
+              run: pip install -e .[dev]
+
+            - name: Run pytest
+              run: pytest -v tests
+              env:
+                  TAG: ${{ matrix.tag }}
+
+            # code coverage
```

Comment:
@yakutovicha I'll let you decide if you want to include the code coverage from here as well. I personally prefer to only to measure the coverage from only the standard unit test. But it is worth to have a try to get the coverage from the notebook smoke test, which will quickly increase the coverage by the black box test cases.

Reply:
I propose to complete this in #441